### PR TITLE
Pull request for reviewing changes on the missing_var branch

### PR DESCRIPTION
**Pull request by billsacks**

### CODE REVIEW

File: cprnc_py/netcdf/scipy/netcdf.py [hunk 1]
```diff
@@ -42,16 +43,13 @@
...
 from numpy.compat import asbytes, asstr
 from numpy import fromstring, dtype, empty, array, asarray
 from numpy import little_endian as LITTLE_ENDIAN
 from functools import reduce
 
-import cprnc_py.netcdf.fs_utils as fs_utils
-
 PY3 = sys.version_info[0] == 3
 
 if PY3:
```
Comment (billsacks) [inline cprnc_py/netcdf/scipy/netcdf.py]: Rather than including this modification of scipy's netcdf.py, I feel it would be easier to maintain this snapshot if we just added a snapshot of the one additional file, scipy._lib.six. At that point, it looks like our netcdf.py would be identical to the scipy version.

Then updating the scipy netcdf.py would just be a matter of pulling in snapshots of these two files - rather than pulling in netcdf.py and merging in the necessary changes (hoping that all of those changes still apply cleanly to the new version, etc.)

File: cprnc_py/netcdf/netcdf_variable_scipy.py [hunk 2]
```diff
@@ -47,11 +47,33 @@ def _get_data_from_slices(self, dim_slices):
         dim_slices should match the dimensionality of this variable
         """
 
-        # FIXME(wjs, 2015-12-31) Will this work on scalar data?
         if len(dim_slices) > 0:
-            vardata = self._var[dim_slices].copy()
+            # TODO(mfd, 2016-09-19) Find a better way of doing this
```
Comment (billsacks) [inline cprnc_py/netcdf/netcdf_variable_scipy.py]: Can you describe the case(s) you're trying to handle with this new code? (My apologies if you already described this to me before, and I'm forgetting.) My initial reaction is to ask if we should really be handling these cases - i.e., it seems that it may actually be correct to raise an exception in these cases rather than trying to handle them. But I don't fully understand what you're doing here, so would be happy to understand this better.

**Comment by mfdeakin-sandia** [comment]
Thank you for the comments, I've made most of the changes, and will verify everything tomorrow.

File: cprnc_py/netcdf/netcdf_variable_scipy.py (hunk 2 above)
Comment (mfdeakin-sandia) [inline cprnc_py/netcdf/netcdf_variable_scipy.py]: I believe that's this commit comment: Verify that the shapes of a variable allow unraveling indexes before attempting to use them
This occurs when the dimension ranges are different for the two variables, which while probably not a standard use case for cprnc, does occur when running on unrelated files.

**Comment by billsacks** [comment]
@mfdeakin-sandia - are you ready for me to re-look at this, or are you still working on the changes? No rush, just wondering where things stand, and if you're waiting for my reply.

**Comment by mfdeakin-sandia** [comment]
Sorry for the delay; I think these changes are ready to be looked at again.

**Comment by billsacks** [comment]
@mfdeakin-sandia - Okay, sorry for the massive delay on my end. I finally got a chance to finish going through this over the weekend. I am basically happy with this, but there were two things I wanted to address (which is what held me up). I'm going to close this PR, and open two PRs into your fork that contain my suggested changes.
